src/fstext/pre-determinize-inl.h

```diff
@@ -236,15 +236,15 @@ inline bool HasBannedPrefixPlusDigits(SymbolTable *symTable, std::string prefix,
   const char *prefix_ptr = prefix.c_str();
   size_t prefix_len = strlen(prefix_ptr);  // allowed to be zero but not encouraged.
   for (SymbolTableIterator siter(*symTable); !siter.Done(); siter.Next()) {
-    const char *sym = siter.Symbol().c_str();
-    if (!strncmp(prefix_ptr, sym, prefix_len)) {  // has prefix.
+    const string &sym = siter.Symbol();
+    if (!strncmp(prefix_ptr, sym.c_str(), prefix_len)) {  // has prefix.
       if (isdigit(sym[prefix_len])) {  // we don't allow prefix followed by a digit, as a symbol.
         // Has at least one digit.
         size_t pos;
         for (pos = prefix_len;sym[pos] != '\0'; pos++)
```

Comment:
Please also change this to 
```suggestion
        for (pos = prefix_len; pos < sym.length(); ++pos)
```
because accessing the string beyond the last character using the `operator[]()` is technically an UB. `length()` is guaranteed O(1) by the C++11 and beyond (and the compiler knows that the string is const in any case).

Reply:
`siter.Symbol()` returns a copy of the string, and this temporary string may be freed at the end of the expression.

C++11 specifies that `str[str.length()]` returns '\0': see [operator\[\]](https://en.cppreference.com/w/cpp/string/basic_string/operator_at). Should I still change this as you suggest?

Reply:
Thanks, I did not know that C++11 defined this! No, no changes required then, LGTM.